graph/task.go

```diff
@@ -197,16 +194,22 @@ func (t *Task) initialize() error {
 
 // UsingRegistryCreds determines whether or not the Task is using registry creds.
 func (t *Task) UsingRegistryCreds() bool {
-	return t.RegistryName != "" &&
-		t.RegistryPassword != "" &&
-		t.RegistryUsername != ""
+	if len(t.Credentials) > 0 {
+		cred := t.Credentials[0]
+		return cred.RegistryName != "" &&
+			cred.RegistryPassword != "" &&
+			cred.RegistryUsername != ""
+	}
+	return false
 }
 
 // SetRegistryInfo sets registry information.
 func (t *Task) setRegistryInfo(registry, user, pw string) {
-	t.RegistryName = registry
-	t.RegistryUsername = user
-	t.RegistryPassword = pw
+	t.Credentials = append(t.Credentials, &Credential{
```

Comment:
where's the struct defined? Make a `NewCredential` function which does validation of all the parameters to ensure they aren't empty; throw an error if they are. Call that method instead of `&Credential`

Reply:
ah, seems like this request did not take in the file I created for that. Doing it.